src/utils/request/xhr.ts

```diff
@@ -141,6 +153,15 @@ export default function request<T>(
       reject(new RequestError(url, xhr.status, "TIMEOUT"));
     };
 
+
+    if (connectionTimeout !== undefined) {
+      xhr.onreadystatechange = function clearConnectionTimeout() {
+        if (xhr.readyState === XMLHttpRequest.HEADERS_RECEIVED) {
```

Comment:
Is it reliably set at the right time for XHR, I'm not too familiar with `onreadystatechange`'s timing.

Reply:
We could also `>=` to catch superior ready states just in case no?

Reply:
`onreadystatechange` with "HEADER_RECEIVED" event is emitted before the first `progress` event.
We can also observe the first `progress` event. In my opinion it's doesn't change much.

You are right for the `>=`